mutation/mutation_compactor.hh

```diff
@@ -629,7 +622,7 @@ public:
         if (!_stop) {
             return {};
         }
-        partition_start ps(std::move(_last_dk), _partition_tombstone);
+        partition_start ps(*std::exchange(_dk, std::nullopt), _partition_tombstone);
```

Comment:
This changes the logic, before the patch we updated _last_dk after consume_end_of_stream(), now we pick up the key before consume_end_of_stream() is called. What is the rationale for this?

Reply:
No change in logic here.
`detach_state()` is called after the read finished a page and the reader is saved in the querier cache. So it is always called after `consume_end_of_stream()`. We relied on this fact to copy `_dk`  just once -- at EOS. But actually this already violates the consume API: `_dk` is not guaranteed to be valid past `consume_end_of_partition()`. Furthermore, the mutation compactor assumed how and when `full_position()` and `detach_state()` is called. Now with these changes, these methods will work correctly regardless of when they are called.

Reply:
I see